program/src/processor.rs

```diff
@@ -643,29 +658,28 @@ impl Processor {
 
         relocate_lamports(
             source_stake_account_info,
-            destination_stake_account_info,
+            destination_info,
             withdraw_lamports,
         )?;
 
         Ok(())
     }
 
     fn process_deactivate(accounts: &[AccountInfo]) -> ProgramResult {
+        let signers = collect_signers(accounts);
         let account_info_iter = &mut accounts.iter();
+
         let stake_account_info = next_account_info(account_info_iter)?;
         let clock_info = next_account_info(account_info_iter)?;
-        let clock = &Clock::from_account_info(clock_info)?;
-        let stake_authority_info = next_account_info(account_info_iter)?;
+        let _option_stake_authority_info = next_account_info(account_info_iter).ok();
```

Comment:
actually maybe it would make more sense if i removed the `.ok()`. this and other similar lines are a result of, i did my best to thoroughly check exactly how many accounts the native stake program asserts exist, and the purpose of the line is just to say "this account may be here, in fact is probably here, although we dont need it directly"

99% of cases it should exist, be a signer, and we collect its pubkey in the `let signers = ...` line. for self-signed stake accounts it may not exist, and since the native program doesnt assert it exists, we cannot unwrap or we break the interface

youre right deleting the line would have the same effect but i think having it (without `.ok()` is better to say "this is what the interface is supposed to be." i think my game plan long term is after we launch a native-compat bpf stake, ill just set up a validator with a geyser plugin to find out whether anyone _actually_ does the things they technically can do, and break anything that is rare or nonexistant in a future update

Reply:
after thinking about this for 30 seconds more i guess a comment would suffice

Reply:
actually there are a couple more of these where the native program fails to assert accounts exist but i assert they do. what im going to do is go through all these again, annotate every function with a comment describing what the native program actually does, and put commented-out lines at the end indicating what extra accounts we would like to be there. we do want comments, rather than actually calling the functions, because even if we dont use them they mutate the accounts iter and so cannot be optimized away by rustc

Reply:
also just noting this is and issue in the realm of "well, lets be as permissive as the native program just in case" and not "oh no we overlooked a major flaw." i honestly dont think many if any of these "weird" transactions exist, since they would require someone to implement their own transaction builders, for no real purpose